lib/get-specs-from-issue.js

```diff
@@ -1,39 +1,117 @@
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
-    context.github.issues.get(...repo, issueNumber),
-    context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
-
-    pushSpecs(specs, allCommentsBody);
-
-    return specs;
+    .then(() => Promise.all([
+      context.github.issues.get(...repo, issueNumber),
+      context.github.issues.listComments(...repo, issueNumber)
+    ]))
+    .then(issueCompleateText => getIssueTextByComments(issueCompleateText))
+    .then(issueTextByComments => getSpecsBlocksByComments(issueTextByComments))
+    .then(issueSpecsBlocks => uniteSpecsBlocks(issueSpecsBlocks))
+    .then(unitedSpecsBlock => getSpecsListItems(unitedSpecsBlock))
+    .then(specListItems => getSpecNamesFromMarkdownList(specListItems));
+}
+
+function getIssueTextByComments(issueCompleateText) {
+  const issueComments = issueCompleateText[0];
+  const issueBody = issueCompleateText[1];
+  const allIssueComments = [];
+
+  allIssueComments.push(issueBody);
+  issueComments.forEach(comment =>
+    allIssueComments.push(comment.body));
+
+  return allIssueComments;
+}
+
+function getSpecsBlocksByComments(completeTexts) {
+  const specBlocks = [];
+  completeTexts.forEach(commentText => {
+    specBlocks.push(getCommentSpecsBlock(commentText));
   });
+
+  return specBlocks;
 }
 
-function pushSpecs(specs, body) {
-  body.toLowerCase();
-  let posiotionOfSpecsList = body.indexOf('## specs');
-  body = body.slice(posiotionOfSpecsList);
-  let endPosiotionOfSpecsList = body.indexOf('##');
-  body = body.slice(0, endPosiotionOfSpecsList);
+function getCommentSpecsBlock(completeText) {
+  const textStartingFromSpecs = findSpecsBlockStart(completeText);
+  return findSpecsBlockEnd(textStartingFromSpecs);
+}
+
+function findSpecsBlockStart(completeText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi;
+  const textLines = completeText.split('\n');
+  var blockStartIndex;
 
-  let bullets = body.split('-');
-  bullets.forEach(bullet => {
-    bullet.trim();
-    if (bullet.startsWith('it')) {
-      specs.push(bullet);
+  _.forEach(textLines, line => {
+    const lineMatchResult = specsBlockStartRegex.exec(line);
+
+    if (lineMatchResult) {
+      specsBlockTopLevelHeader = lineMatchResult[1].length;
+      blockStartIndex = completeText.indexOf(line);
     }
   });
+
+  return completeText.slice(blockStartIndex);
+}
+
+function findSpecsBlockEnd(textStartingFromSpecs) {
+  const markdownTitleRegex = /(#+)/g;
+  const textLines = textStartingFromSpecs.split('\n');
+  var blockEndIndex;
+
+  _.forEach(textLines, line => {
+    const lineMatchResult = markdownTitleRegex.exec(line);
+    if (lineMatchResult && lineMatchResult[1]) {
+      var titleLevel = lineMatchResult[1].length;
+    }
+    if (specsBlockTopLevelHeader <= titleLevel) {
```

Comment:
Wouldn't that be an issue when:
```
line = '####';
topLevel = '##';
currentTitleLevel.startsWith(topLevel) // this is true
```

Reply:
no, because `#### some text` is a lower level header than `## Specs`.

`topLevel` is an `h2`, which is a second order title, while `line` is an `h4`, which is a fourth in order title.

so `line` is part of the block in the example you gave.

The last check should be performed on `line`

Reply:
what's the algorithmic difference between what you implemented and what I suggested? :-)